ometiff-JavaScript/PixelSizeConsistency.js

```diff
@@ -1,30 +1,56 @@
 }
+
+function convertFactor(physicalSizeUnit) {
+    if(physicalSizeUnit === 'mm') {
+        return 1000
+    }else if(physicalSizeUnit === 'um') {
+        return 1
+    }else if(physicalSizeUnit === 'nm') {
+        return 0.001
+    }
+}
+
 getExif().then(function(result) {
     var xml = result['ImageDescription']['description']
     //console.log("\n OME-XML:", xml)
     
+    parseString(xml, function (err, output) {
 
-    var parseString = xml2js.parseString;
+        const physicalSizeX = output['OME']['Image'][0]['Pixels'][0]['$']['PhysicalSizeX']
+        const physicalSizeXUnit = output['OME']['Image'][0]['Pixels'][0]['$']['PhysicalSizeXUnit']
+        const physicalSizeY = output['OME']['Image'][0]['Pixels'][0]['$']['PhysicalSizeY']
+        const physicalSizeYUnit = output['OME']['Image'][0]['Pixels'][0]['$']['PhysicalSizeYUnit']
+        const physicalSizeZ = output['OME']['Image'][0]['Pixels'][0]['$']['PhysicalSizeZ']
+        const physicalSizeZUnit = output['OME']['Image'][0]['Pixels'][0]['$']['PhysicalSizeZUnit']
 
-    parseString(xml, function (err, output) {
-        //console.dir(output['OME']['Image'][0]['Pixels'][0]['$'])
         console.log("\nOME-TIFF metadata\n---------------------")
-        console.log("PhysicalSizeX:", output['OME']['Image'][0]['Pixels'][0]['$']['PhysicalSizeX']);
-        console.log("PhysicalSizeXUnit:", output['OME']['Image'][0]['Pixels'][0]['$']['PhysicalSizeXUnit']);
-        console.log("PhysicalSizeY:", output['OME']['Image'][0]['Pixels'][0]['$']['PhysicalSizeY']);
-        console.log("PhysicalSizeYUnit:", output['OME']['Image'][0]['Pixels'][0]['$']['PhysicalSizeYUnit']);
-        console.log("PhysicalSizeZ:", output['OME']['Image'][0]['Pixels'][0]['$']['PhysicalSizeZ']);
-        console.log("PhysicalSizeZUnit:", output['OME']['Image'][0]['Pixels'][0]['$']['PhysicalSizeZUnit']);
-        console.log("DimensionOrder:", output['OME']['Image'][0]['Pixels'][0]['$']['DimensionOrder']);
-    });
-})
+        console.log("PhysicalSizeX:", physicalSizeX)
+        console.log("PhysicalSizeXUnit:", physicalSizeXUnit)
+        console.log("PhysicalSizeY:", physicalSizeY)
+        console.log("PhysicalSizeYUnit:", physicalSizeYUnit)
+        console.log("PhysicalSizeZ:", physicalSizeZ)
+        console.log("PhysicalSizeZUnit:", physicalSizeZUnit)
+        console.log("DimensionOrder:", output['OME']['Image'][0]['Pixels'][0]['$']['DimensionOrder'])
+
+        let factorX = convertFactor(physicalSizeXUnit);
+        let factorY = convertFactor(physicalSizeYUnit);
+        let factorZ = convertFactor(physicalSizeZUnit);
+        let jsonData = require('./test_image.json')
+
+        let pixelSize = jsonData['PixelSize']
 
-//fetch("test_image.json")
-//  .then(response => response.json())
-//  .then(json => console.log(json));
\ No newline at end of file
+        if (physicalSizeX * factorX !== pixelSize[0] || physicalSizeY * factorY !== pixelSize[1] || physicalSizeZ * factorZ !== pixelSize[2]) {
```

Comment:
The original `PixelSize` field in the JSON file was always in `um`. However, after community review, we also have a `PixelSizeUnits` field that can be either `mm`, `um` or `nm`. So the conversion factors would need to take into account both the unit of the JSON file and the OMETIFF file.

Reply:
Done